### PR TITLE
OpenShift Pipelines release notes 1.17 (upstream)

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `pipelines-docs-1.17`

Issue: https://issues.redhat.com/browse/RHDEVDOCS-6211

Link to docs preview:  https://86324--ocpdocs-pr.netlify.app/openshift-pipelines/latest/release_notes/op-release-notes-1-17.html#op-release-notes-1-17_op-release-notes

QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/op-release-notes-1-17.adoc [hunk 1]
```diff
@@ -192,3 +196,27 @@ For more information about using the tasks in the `openshift-pipelines` namespac
 * Before this update, running the `tkn pac create repo` command on an empty repository resulted in the `.` period symbol being generated as the name of the pipeline run in the template. With this update, the issue is fixed, and the `REPO_NAME.git` is now used as the pipeline run name.
 
 * With this update, the pipeline run failure handling is fixed to accurately report validation failures in the `status.message` field and consistently trigger `finally` tasks, even if a task fails the validation.
+
+* Before this update, when a pod failed because of an out-of-memory error, the task run did not immediately fail. Instead, it was unresponsive for some time and then failed. With this update, the task run immediately fails.
```
Comment (mramendi) [inline modules/op-release-notes-1-17.adoc:200]: pipeline section not PAC

File: modules/op-release-notes-1-17.adoc [hunk 2]
```diff
@@ -192,3 +196,27 @@ For more information about using the tasks in the `openshift-pipelines` namespac
...
+
+* Before this update, if a `CustomRun` resource referred to a pipeline or task resource that returned an array as a result, a result type mismatch was reported in the log. With this update, the array result is processed correctly.
+
+* Before this update, when a task run failed, in some cases the statuses of some of the steps in the task were not recorded correctly in the `TaskRun` YAML manifesto. With this update, the statuses are correctly recorded.
```
Comment (mramendi) [inline modules/op-release-notes-1-17.adoc:204]: pipeline section not PAC

File: modules/op-release-notes-1-17.adoc [hunk 3]
```diff
@@ -192,3 +196,27 @@ For more information about using the tasks in the `openshift-pipelines` namespac
...
 
 * With this update, the pipeline run failure handling is fixed to accurately report validation failures in the `status.message` field and consistently trigger `finally` tasks, even if a task fails the validation.
+
+* Before this update, when a pod failed because of an out-of-memory error, the task run did not immediately fail. Instead, it was unresponsive for some time and then failed. With this update, the task run immediately fails.
+
+* Before this update, if a `CustomRun` resource referred to a pipeline or task resource that returned an array as a result, a result type mismatch was reported in the log. With this update, the array result is processed correctly.
+
+* Before this update, when a task run failed, in some cases the statuses of some of the steps in the task were not recorded correctly in the `TaskRun` YAML manifesto. With this update, the statuses are correctly recorded. 
+
+* Before this update, if a step in a task failed, the subsequent steps were sometimes not marked as skipped in the YAML manifest of the task run or pipeline run. With this update, the steps are marked as skipped.
```
Comment (mramendi) [inline modules/op-release-notes-1-17.adoc:206]: pipeline section not PAC

**Review by vdemeester — APPROVED** [review]

**Review by chmouel — APPROVED** [review]

**Review by ppitonak — APPROVED** [review]

**Review by GroceryBoyJr — COMMENTED** [review]
One small nit! NOTE: There are errors in the text that are not part of this PR, you might consider resolving. Line 199 {piplines-shortname} and line 150 (the the)  but are not part of this PR.

File: modules/op-release-notes-1-17.adoc [hunk 4]
```diff
@@ -94,6 +94,10 @@ spec:
 # ...
 ----
 
+* With this update, after the execution of a pipeline run or task run, the YAML manifest of the resource contains a label showing the ID of the user who executed it. For a `PipelineRun` resource, the label is `tekton.dev/PipelineRunUID`. For a `TaskRun` resource, the label is `tekton.dev/TaskRunUID`.
+
+* With this update, when you use the `spec.pipeline,performance` spec in the `TektonConfig` CR to specify performance tuning values such as `threads-per-controller`, `kube-api-qps`, and `kube-api-burst`, {pipelines-shortname} applies these values to the resolver controller as well as the other controllers.
```
Comment (bergerhoffer) [inline modules/op-release-notes-1-17.adoc:99]: Is this a typo?

```suggestion
* With this update, when you use the `spec.pipeline.performance` spec in the `TektonConfig` CR to specify performance tuning values such as `threads-per-controller`, `kube-api-qps`, and `kube-api-burst`, {pipelines-shortname} applies these values to the resolver controller as well as the other controllers.
```

File: modules/op-release-notes-1-17.adoc [hunk 5]
```diff
@@ -105,17 +109,45 @@ spec:
...
 === {pac}
 
 * With this update, the `pipelines_as_code_running_pipelineruns_count` metric is added to show the number of running `PipelineRun` resources in {pac}. The metric can be filtered by repository or by namespace.
 
 * With this update, the `pipelines_as_code_pipelinerun_duration_seconds_sum` metric is added to show the total duration in seconds taken by all `PipelineRun` resources in {pac}. The metric can be filtered by repository, namespace, `PipelineRun` status, and `PipelineRun` status change reason.
 
+[id="tekton-results-new-features-1-17_{context}"]
+=== {tekton-results}
+
+* With this update, you can configure {tekton-results} to include certain labels and annotation from the YAML manifests of pipelines and tasks in the `summary fields` column in the results tables. By default, the value of the `tekton.dev/pipeline` label is used. To make this change, modify the `TektonResult` CR as shown in the following example:
```
Comment (bergerhoffer) [inline modules/op-release-notes-1-17.adoc:124]:
```suggestion
* With this update, you can configure {tekton-results} to include certain labels and annotations from the YAML manifests of pipelines and tasks in the `summary fields` column in the results tables. By default, the value of the `tekton.dev/pipeline` label is used. To make this change, modify the `TektonResult` CR as shown in the following example:
```

File: modules/op-release-notes-1-17.adoc [hunk 6]
```diff
@@ -105,17 +109,45 @@ spec:
...
++
+.Example of configuring summary labels and annotations in the `TektonResult` CR
+[source,yaml]
+----
+apiVersion: operator.tekton.dev/v1
+kind: TektonResult
+metadata:
+  name: result
+spec:
+     options:
```
Comment (bergerhoffer) [inline modules/op-release-notes-1-17.adoc:134]: Indentation seems off here, just checking if that's okay

File: modules/op-release-notes-1-17.adoc [hunk 7]
```diff
@@ -192,3 +224,39 @@ For more information about using the tasks in the `openshift-pipelines` namespac
 * Before this update, running the `tkn pac create repo` command on an empty repository resulted in the `.` period symbol being generated as the name of the pipeline run in the template. With this update, the issue is fixed, and the `REPO_NAME.git` is now used as the pipeline run name.
 
 * With this update, the pipeline run failure handling is fixed to accurately report validation failures in the `status.message` field and consistently trigger `finally` tasks, even if a task fails the validation.
+
+* Before this update, when a pod failed because of an out-of-memory error, the task run did not immediately fail. Instead, it was unresponsive for some time and then failed. With this update, the task run immediately fails.
+
+* Before this update, if a `CustomRun` resource referred to a pipeline or task resource that returned an array as a result, a result type mismatch was reported in the log. With this update, the array result is processed correctly.
+
+* Before this update, when a task run failed, in some cases the statuses of some of the steps in the task were not recorded correctly in the `TaskRun` YAML manifesto. With this update, the statuses are correctly recorded.
```
Comment (bergerhoffer) [inline modules/op-release-notes-1-17.adoc:232]:
```suggestion
* Before this update, when a task run failed, in some cases the statuses of some of the steps in the task were not recorded correctly in the `TaskRun` YAML manifest. With this update, the statuses are correctly recorded. 
```

**Review by bergerhoffer — COMMENTED** [review]
Couple small typos

**Comment by bergerhoffer** [comment]
Per discussion in slack with @mramendi, will merge this PR as is, and then he'll do a separate PR to fix the typos

**Pull request merged**